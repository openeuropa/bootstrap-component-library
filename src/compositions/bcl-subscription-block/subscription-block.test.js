import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import blockData from "@openeuropa/bcl-subscription-block/data/data_block.js";

const template = "@oe-bcl/bcl-subscription-block/subscription-block.html.twig";

const render = (params) => renderTwigFileAsNode(template, params, true);

describe("OE - Subscription Block", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(blockData)).resolves.toMatchSnapshot();
  });
});
