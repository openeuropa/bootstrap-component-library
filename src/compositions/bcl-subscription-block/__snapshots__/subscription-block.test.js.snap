// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Subscription Block renders correctly 1`] = `
<jest>
  <div
    class="bg-light px-4-5 pt-4 pb-4-5"
  >
    <div
      class="fs-2 fw-normal"
    >
      Subscribe
    </div>
    <p
      class="fw-bold"
    >
      Get notified lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
    <div
      class="pb-2"
    >
      <button
        class="btn btn-secondary btn-lg"
        data-bs-target="#subscribeModal"
        data-bs-toggle="modal"
        type="button"
      >
        Subscribe
        <svg
          class="ms-2-5 bi icon--fluid"
        >
          <use
            xlink:href="/icons.svg#chevron-right"
          />
        </svg>
      </button>
    </div>
  </div>
</jest>
`;
